core/src/main/java/com/sequenceiq/cloudbreak/repository/RdsConfigWithoutClusterRepository.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.repository;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.transaction.Transactional;
+import javax.transaction.Transactional.TxType;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.ResourceStatus;
+import com.sequenceiq.cloudbreak.domain.RDSConfig;
+import com.sequenceiq.cloudbreak.domain.view.RdsConfigWithoutCluster;
+import com.sequenceiq.cloudbreak.workspace.repository.EntityType;
+
+@EntityType(entityClass = RDSConfig.class)
+@Transactional(TxType.REQUIRED)
+public interface RdsConfigWithoutClusterRepository extends CrudRepository<RDSConfig, Long> {
```

Comment:
General note: I wonder how this is able to return `RdsConfigWithoutCluster` instances while operating on the entity type `RDSConfig`, without the latter explicitly implementing `RdsConfigWithoutCluster`. I suppose the returned object has a dynamically constructed class type that actually implements `RdsConfigWithoutCluster` via CGLIB etc. magic.

Reply:
Nothing magic behind the solution. We are 2 choices for implementing the view:
* create a copy of the entity with fields, getters and setters and annotation -> you cannot ignore the references because it should exist for the where clause. unfortunately, the hibernate will fetch the reference if any reflection magic or string or any magic call (the cluster references here) 
* create an interface with just getter and querying with the original entity but projection the requested field -> you can ignore the references because the column exists in the entity you can write the where clause but the hibernate never fetch it